kv/kvserver: TestInitRaftGroupOnRequest failed

kv/kvserver.TestInitRaftGroupOnRequest [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5021194&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5021194&tab=artifacts#/) on master @ [2fdc0a62df08829dd96901fffb5fb1fbb0b0997d](https://github.com/cockroachdb/cockroach/commits/2fdc0a62df08829dd96901fffb5fb1fbb0b0997d):


```
=== RUN   TestInitRaftGroupOnRequest
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/751d67000aac5f3394c2369309253f02/logTestInitRaftGroupOnRequest3240201838
    test_log_scope.go:80: use -show-logs to present logs inline
    client_raft_test.go:4259: [NotLeaseHolderError] lease acquisition attempt lost to another lease, which has expired in the meantime; r45: replica (n2,s2):2LEARNER not lease holder; lease holder unknown
    panic.go:642: -- test log scope end --
--- FAIL: TestInitRaftGroupOnRequest (5.15s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitRaftGroupOnRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-15579

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -4263,15 +4263,13 @@ func TestInitRaftGroupOnRequest(t *testing.T) {
 		log.Errorf(ctx, "expected raft group to be uninitialized")
 	}
 	// Send an increment and verify that initializes the Raft group.
-	_, pErr := kv.SendWrapped(ctx,
-		followerStore.TestSender(), incrementArgs(splitKey, 1))
-	if pErr != nil {
-		t.Fatal(pErr)
-	}
+	//
+	// NB: We don't know who has the lease, so we ignore any errors (i.e.
+	// NotLeaseHolderError). We only care that it initializes the Raft group.
+	_, pErr := kv.SendWrapped(ctx, followerStore.TestSender(), incrementArgs(splitKey, 1))
+	_ = pErr // appease returncheck linter
 
-	if !repl.IsRaftGroupInitialized() {
-		t.Fatal("expected raft group to be initialized")
-	}
+	require.True(t, repl.IsRaftGroupInitialized(), "expected raft group to be initialized")
 }
 
 // TestFailedConfChange verifies correct behavior after a configuration change

